rust/src/storage/config.rs

```diff
@@ -31,6 +30,65 @@ use object_store::gcp::{GoogleCloudStorageBuilder, GoogleConfigKey};
 ))]
 use std::str::FromStr;
 
+/// Recognises various URL formats, identifying the relevant [`ObjectStore`](crate::ObjectStore)
+#[derive(Debug, Eq, PartialEq)]
+enum ObjectStoreScheme {
+    /// Url corresponding to LocalFileSystem
+    Local,
+    /// Url corresponding to InMemory
+    Memory,
+    /// Url corresponding to S3
+    AmazonS3,
+    /// Url corresponding to GoogleCloudStorage
+    GoogleCloudStorage,
+    /// Url corresponding to MicrosoftAzure
+    MicrosoftAzure,
+    /// Url corresponding to HttpStore
+    Http,
+    /// Url corresponding to Hdfs
+    Hdfs,
+}
+
+impl ObjectStoreScheme {
+    /// Create an [`ObjectStoreScheme`] from the provided [`Url`]
+    ///
+    /// Returns the [`ObjectStoreScheme`] and the remaining [`Path`]
+    fn parse(url: &Url) -> Result<(Self, Path), ObjectStoreError> {
+        let strip_bucket = || Some(url.path().strip_prefix('/')?.split_once('/')?.1);
+
+        let (scheme, path) = match (url.scheme(), url.host_str()) {
+            ("file", None) => (Self::Local, url.path()),
+            ("memory", None) => (Self::Memory, url.path()),
+            ("s3" | "s3a", Some(_)) => (Self::AmazonS3, url.path()),
+            ("gs", Some(_)) => (Self::GoogleCloudStorage, url.path()),
+            ("az" | "adl" | "azure" | "abfs" | "abfss", Some(_)) => {
+                (Self::MicrosoftAzure, url.path())
+            }
+            ("http", Some(_)) => (Self::Http, url.path()),
+            ("hdfs", Some(_)) => (Self::Hdfs, url.path()),
+            ("https", Some(host)) => {
+                if host.ends_with("dfs.core.windows.net") || host.ends_with("blob.core.windows.net")
+                {
```

Comment:
this does not seem to have the latest object store changes for fabric support

Reply:
will definately add that !